arrows/vxl/high_pass_filter.cxx

```diff
@@ -0,0 +1,490 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include "high_pass_filter.h"
+
+#include <arrows/vxl/image_container.h>
+#include <vital/util/enum_converter.h>
+
+#include <vil/vil_convert.h>
+#include <vil/vil_image_view.h>
+#include <vil/vil_math.h>
+#include <vil/vil_plane.h>
+#include <vil/vil_transpose.h>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace vxl {
+
+enum filter_mode
+{
+  MODE_box,
+  MODE_bidir,
+};
+
+ENUM_CONVERTER( mode_converter, filter_mode, { "box", MODE_box }, { "bidir",
+                                                                    MODE_bidir } )
+
+// --------------------------------------------------------------------------------------
+/// Private implementation class
+class high_pass_filter::priv
+{
+public:
+
```

Comment:
I'd avoid the blank line here. However, while we don't "typically", given the number and size of member functions with definitions inside the class definition, separator lines might be helpful:
```suggestion
  // --------------------------------------------------------------------------
```
(...and before other methods.)

Reply:
This looked quite messy to me so I pulled the definitions outside of the class definition.